x-pack/plugins/infra/public/components/metrics_explorer/chart_context_menu.tsx

```diff
@@ -0,0 +1,202 @@
+}
+
+const fieldToNodeType = (
+  source: SourceConfiguration,
+  field: string
+): InventoryItemType | undefined => {
+  if (source.fields.host === field) {
+    return 'host';
+  }
+  if (source.fields.pod === field) {
+    return 'pod';
+  }
+  if (source.fields.container === field) {
+    return 'container';
+  }
+};
+
+const dateMathExpressionToEpoch = (dateMathExpression: string, roundUp = false): number => {
+  const dateObj = DateMath.parse(dateMathExpression, { roundUp });
+  if (!dateObj) throw new Error(`"${dateMathExpression}" is not a valid time string`);
+  return dateObj.valueOf();
+};
+
+export const createNodeDetailLink = (
+  nodeType: InventoryItemType,
+  nodeId: string,
+  from: string,
+  to: string,
+  prefixPathWithBasePath: (app: string, path?: string | undefined) => string | undefined
+) => {
+  return getNodeDetailUrl({
+    nodeType,
+    nodeId,
+    from: dateMathExpressionToEpoch(from),
+    to: dateMathExpressionToEpoch(to, true),
+    prefixPathWithBasePath,
+  });
+};
+
+export const MetricsExplorerChartContextMenu = ({
+  onFilter,
+  options,
+  series,
+  source,
+  timeRange,
+  uiCapabilities,
+  chartOptions,
+}: Props) => {
+  const getUrlForApp = useKibana().services.application?.getUrlForApp;
+  const prependBasePath = useKibana().services.http?.basePath.prepend;
+  const prefixPathWithBasePath = useCallback(
+    (app: string, path?: string) => {
+      if (!getUrlForApp || !prependBasePath) {
+        return;
+      }
+      return prependBasePath(getUrlForApp(app, { path }));
+    },
+    [getUrlForApp, prependBasePath]
+  );
+  const [isPopoverOpen, setPopoverState] = useState(false);
+  const supportFiltering = options.groupBy != null && onFilter != null;
+  const handleFilter = useCallback(() => {
+    // onFilter needs check for Typescript even though it's
+    // covered by supportFiltering variable
+    if (supportFiltering && onFilter) {
+      onFilter(`${options.groupBy}: "${series.id}"`);
+    }
+    setPopoverState(false);
+  }, [supportFiltering, options.groupBy, series.id, onFilter]);
+
+  const tsvbUrl = createTSVBLink(source, options, series, timeRange, chartOptions);
+
+  // Only display the "Add Filter" option if it's supported
+  const filterByItem = supportFiltering
+    ? [
+        {
+          name: i18n.translate('xpack.infra.metricsExplorer.filterByLabel', {
+            defaultMessage: 'Add filter',
+          }),
+          icon: 'metricsApp',
+          onClick: handleFilter,
+          'data-test-subj': 'metricsExplorerAction-AddFilter',
+        },
+      ]
+    : [];
+
+  const nodeType = source && options.groupBy && fieldToNodeType(source, options.groupBy);
+  const viewNodeDetail = nodeType
+    ? [
+        {
+          name: i18n.translate('xpack.infra.metricsExplorer.viewNodeDetail', {
+            defaultMessage: 'View metrics for {name}',
+            values: { name: nodeType },
+          }),
+          icon: 'metricsApp',
+          href: createNodeDetailLink(
```

Comment:
The generated URL contains the space prefix twice:

![grafik](https://user-images.githubusercontent.com/973741/74457558-b35da900-4e88-11ea-866b-c46082b0a18a.png)

Reply:
👍 Fixed with `prefixPathWithBasepath` changes.